x-pack/plugins/enterprise_search/server/lib/enterprise_search_config.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { readFileSync } from 'fs';
+
+import http from 'http';
+import https from 'https';
+
+import { schema, TypeOf } from '@kbn/config-schema';
+
+// -----------------------------------------------------------------------------
+export const configSchema = schema.object({
+  host: schema.maybe(schema.string()),
+  enabled: schema.boolean({ defaultValue: true }),
+  accessCheckTimeout: schema.number({ defaultValue: 5000 }),
+  accessCheckTimeoutWarning: schema.number({ defaultValue: 300 }),
+  ssl: schema.object({
+    certificateAuthorities: schema.maybe(
+      schema.oneOf([schema.arrayOf(schema.string()), schema.string()])
+    ),
+    rejectUnauthorized: schema.boolean({ defaultValue: false }),
+  }),
+});
+
+type EnterpriseSearchConfigType = TypeOf<typeof configSchema>;
+
+export const config: PluginConfigDescriptor<EnterpriseSearchConfigType> = {
```

Comment:
You missed the type import for this it looks like:

```js
import { PluginConfigDescriptor } from 'src/core/server';
```

You'll also need to update `server/index.ts` to remove the now-unused `PluginConfigDescriptor` import - an unused import lint check will fail